src/test/java/com/lecture/jpa/domain/CustomerRepositoryTest.java

```diff
@@ -0,0 +1,102 @@
+package com.lecture.jpa.domain;
+
+import jakarta.transaction.Transactional;
+import org.assertj.core.util.Lists;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
+@Transactional
+class CustomerRepositoryTest {
+
+    @Autowired
+    CustomerRepository customerRepository;
+
+    @AfterEach
+    void tearDown() {
+        customerRepository.deleteAll();
+    }
+
+    @Test
+    void 고객정보가_저장되는지_확인한다() {
+        // Given
+        Customer customer = new Customer();
+        customer.setId(1L);
+        customer.setFirstName("beomu");
+        customer.setLastName("kim");
+
+        // When
+        customerRepository.save(customer);
+
+        // Then
+        Customer selectedEntity = customerRepository.findById(1L).get();
+        assertThat(selectedEntity.getId()).isEqualTo(1L);
+        assertThat(selectedEntity.getFirstName()).isEqualTo(customer.getFirstName());
+    }
+
+    @Test
+    void 고객정보가_수정되는지_확인한다() {
+        // Given
+        Customer customer = new Customer();
+        customer.setId(1L);
+        customer.setFirstName("beomu");
+        customer.setLastName("kim");
+        Customer entity = customerRepository.save(customer);
+
+        // When
+        entity.setFirstName("heungmin");
+        entity.setLastName("son");
+
+        // Then
+        Customer selectedEntity = customerRepository.findById(1L).get();
+        assertThat(selectedEntity.getId()).isEqualTo(1L);
```

Comment:
테스트를 작성할때 하나의 테스트 메서드에는 하나의 단언(assertion)만 있는것이 좋다고 학습한 경험이 있어요.
```suggestion
        assertThat(selectedEntity )
        .usingRecursiveComparison()
        .isEqualTo(entity);
```
위 메서드 한번 사용해보시는것 어떨까요?

Reply:
그러면 자연스럽게 getter도 사용하지 않을 수 있을것 같아요

Reply:
usingRecursiveComparison는 처음 보네요! 한번 알아볼게요 감사합니다

Reply:
`hamcrest`의 `samePropertyValuesAs`도 있는데 참고하시면 좋을 것 같아요!
저는 hamcrest의 메서드 명이 조금 더 직관적이라 많이 쓰는 것 같아요 ㅎㅎ